implementation/core/src/test/java/io/smallrye/faulttolerance/core/bulkhead/CompletionStageThreadPoolBulkheadTest.java

```diff
@@ -226,42 +221,31 @@ public void shouldNotStartNextIfCSInProgress() throws Exception {
         party.organizer().waitForAll();
 
         CompletionStage<String> secondResult = bulkhead.apply(new InvocationContext<>(null));
-        waitUntilQueueSize(bulkhead, 1, 100);
+        waitUntilQueueSize(bulkhead, 1, 150);
 
         party.organizer().disband();
 
         assertThat(firstResult.toCompletableFuture().get()).isEqualTo("shouldNotStartNextIfCSInProgress");
         assertThat(secondResult.toCompletableFuture().get()).isEqualTo("shouldNotStartNextIfCSInProgress");
-        assertThat(bulkhead.getQueueSize()).isEqualTo(0);
+        assertThat(bulkhead.getQueueSize()).isZero();
     }
 
-    // TODO waiting for a condition in a unit test shouldn't really be needed
-    //  ultimately, we should use Awaitility for waiting for a condition in a test, not home-grown utils like this
-    private static <V> void waitUntilQueueSize(CompletionStageThreadPoolBulkhead<V> bulkhead, int size, long timeout)
-            throws Exception {
-        long start = System.currentTimeMillis();
-        while (System.currentTimeMillis() - start < timeout) {
-            Thread.sleep(50);
-            if (bulkhead.getQueueSize() == size) {
-                return;
-            }
-        }
-        fail("queue not reached size " + size + " in " + timeout + " [ms]");
+    private static <V> void waitUntilQueueSize(CompletionStageThreadPoolBulkhead<V> bulkhead, int size, long timeout) {
+        await().atMost(Duration.ofMillis(timeout)).until(() -> bulkhead.getQueueSize() == size);
+    }
+
+    private static <V> CompletionStage<V> getSingleFinishedResult(List<CompletionStage<V>> results, long timeout) {
```

Comment:
This split to 2 methods looks strange at first. It's probably fine, it just made me stop for a bit.

Reply:
I preferred to split to increase readability. I found my initial non splitted version pretty polluted.

    private static <V> CompletionStage<V> getSingleFinishedResult(List<CompletionStage<V>> results, long timeout) {
        return await().atMost(Duration.ofMillis(timeout))
                      .<Optional<CompletionStage<V>>>until(() -> {
                          for (CompletionStage<V> result : results) {
                              if (result.toCompletableFuture().isDone()) {
                                  return Optional.of(result);
                              }
                          }
                          return Optional.empty();
                      }, Optional::isPresent)
                      .get();
    }

WDYT?

Reply:
Yea it's probably fine. I didn't even know `until` can return a value :-)

Reply:
Me neither. I just found out :)